rpc/src/rpc_subscriptions.rs

```diff
@@ -1139,179 +654,133 @@ impl RpcSubscriptions {
+                    );
+                    num_signatures_found += 1;
+                    if notified {
+                        num_signatures_notified += 1;
+                    }
+                }
+                _ => error!("wrong subscription type in alps map"),
+            }
         }
-        signatures_time.stop();
-        let total_notified = num_pubkeys_notified + num_programs_notified + num_signatures_notified;
-        let total_ms = accounts_time.as_ms() + programs_time.as_ms() + signatures_time.as_ms();
+
+        total_time.stop();
+
+        let total_notified = num_accounts_notified
+            + num_logs_notified
+            + num_programs_notified
+            + num_signatures_notified;
+        let total_ms = total_time.as_ms();
         if total_notified > 0 || total_ms > 10 {
             debug!(
-                "notified({}): accounts: {} / {} ({}) programs: {} / {} ({}) signatures: {} / {} ({})",
+                "notified({}): accounts: {} / {} logs: {} / {} programs: {} / {} signatures: {} / {}",
                 source,
-                pubkeys.len(),
-                num_pubkeys_notified,
-                accounts_time,
-                programs.len(),
+                num_accounts_found,
+                num_accounts_notified,
+                num_logs_found,
+                num_logs_notified,
+                num_programs_found,
                 num_programs_notified,
-                programs_time,
-                signatures.len(),
+                num_signatures_found,
                 num_signatures_notified,
-                signatures_time,
             );
             inc_new_counter_info!("rpc-subscription-notify-bank-or-gossip", total_notified);
             datapoint_info!(
                 "rpc_subscriptions",
                 ("source", source.to_string(), String),
-                ("num_account_subscriptions", pubkeys.len(), i64),
-                ("num_account_pubkeys_notified", num_pubkeys_notified, i64),
-                ("accounts_time", accounts_time.as_us() as i64, i64),
-                ("num_logs_subscriptions", logs.len(), i64),
+                ("num_account_subscriptions", num_accounts_found, i64),
+                ("num_account_pubkeys_notified", num_accounts_notified, i64),
+                ("num_logs_subscriptions", num_logs_found, i64),
                 ("num_logs_notified", num_logs_notified, i64),
-                ("logs_time", logs_time.as_us() as i64, i64),
-                ("num_program_subscriptions", programs.len(), i64),
+                ("num_program_subscriptions", num_programs_found, i64),
                 ("num_programs_notified", num_programs_notified, i64),
-                ("programs_time", programs_time.as_us() as i64, i64),
-                ("num_signature_subscriptions", signatures.len(), i64),
+                ("num_signature_subscriptions", num_signatures_found, i64),
                 ("num_signatures_notified", num_signatures_notified, i64),
```

Comment:
```suggestion
                ("num_signatures_notified", num_signatures_notified, i64),
                ("notifications_time", total_time.as_us() as i64, i64),
```

Reply:
Hey, I've added `num_*_subscribers` metrics that represent the number of references to tokens of a certain subscription kind. That means though if a client is subscribed to `n` different subscriptions of a same kind, they will be count as `n` separate subscribers.

If you meant counting the amount of clients that *participate* in a certain subscription kind, then it's not that easy and would likely require a separate struct (maybe even two) with some atomic counters, which might seems like an overkill for a metric. But still doable.

`notifications_time` is added as well.

Reply:
Thank you!
Yes, I did mean number of clients that would receive notifications of a certain subscription kind, so that it would be parallel to the old `num_accounts_notified`/`num_signatures_notified`/`num_programs_notified`. It would be a lot easier to compare apples to apples if we had that. I'll ponder whether it's worth adding atomics for that, though.

In the meantime, any word on how `notifications_time` compares to the old `accounts_time` metric in your 40000+ simultaneous subscriptions test?